examples/nlu-assistant/README.md

```diff
@@ -0,0 +1,507 @@
+
+- `Greeting.txt`
+- `Joke.txt`
+- `GetDirections.txt`
+- `Weather.txt`
+- `Thanks.txt`
+- `Farewell.txt`
+
+Now, inside every one of them, add several examples as follows:
+
+**src/nlu/utterances/en/GetDirections.txt**
+
+```
+how can i reach a train station?
+get directions to the nearest ATM
+how do I get home taking E120
+find the fastest route to Baxter Building 42nd street Madison Avenue Manhattan
+how to get to the Main Street avoiding heavy traffic
+...
+```
+
+Each example should be defined in single line and should be separated of the next one by a newline.
+
+> **Note:** Take into account that in order to train deep learning models, the more example the better your neural network will perform. For the sake of this example we have added 50 examples per intent, but they could be insufficient.
+
+Once you have defined all the examples for every intent, we are going to proceed with the model creation.
+
+#### 2. Creating your Neural Network model
+Under **src/nlu/** you will notice that there is a `train-model.ts` file. This is the main entry point responsible of running your model.
+
+##### 2.1 Understanding the main points of a model
+
+We could start by using the example which comes with the `nlu` example:
+
+```ts
+import { BotonicNLU, ModelTemplatesType } from '@botonic/nlu'
+import { join } from 'path'
+
+const LANGUAGE = 'en'
+const UTTERANCES_PATH = join(
+  process.cwd(),
+  'src',
+  'nlu',
+  'utterances',
+  LANGUAGE
+)
+
+const nlu = new BotonicNLU()
+const data = nlu.readData({
+  path: UTTERANCES_PATH,
+  language: LANGUAGE,
+  maxSeqLen: 20,
+})
+
+const [xTrain, xTest, yTrain, yTest] = nlu.trainTestSplit({
+  data: data,
+  testPercentage: 0.1,
+})
+
+;(async () => {
+  await nlu.createModel({
+    template: ModelTemplatesType.SIMPLE_NN,
+    learningRate: 5e-3,
+    trainableEmbeddings: true,
+  })
+  await nlu.train(xTrain, yTrain, { epochs: 8 })
+  await nlu.saveModel()
+  console.log('Accuracy:', nlu.evaluate(xTest, yTest))
+  console.log('Model saved.')
+})()
+```
+
+Let's break it down into little pieces:
+
+1. First of all, you need to define which is the target language to be trained. You must specify both the language and the directory from where the utterances will be retrieved:
+   ```ts
+   const LANGUAGE = 'en'
+   const UTTERANCES_PATH = join(
+     process.cwd(),
+     'src',
+     'nlu',
+     'utterances',
+     LANGUAGE
+   )
+   ```
+2. Now we will initialize an instance of `BotonicNLU` and feed it with our data:
+   ```ts
+   const nlu = new BotonicNLU()
+   const data = nlu.readData({
+     path: UTTERANCES_PATH,
+     language: LANGUAGE,
+     maxSeqLen: 20,
+   })
+   ```
+   `maxSeqLen` defines the maximum length of the sentences to train the NN. As webchat is a messaging channel, a `maxSeqLen` of 20 will be enough to cover our needs (_it's pretty uncommon that a user types an input of more of 20 words_).
```

Comment:
maxSeqLen do not represent a limitation for the length of the sentences, it represents how long will be the sequence inputs:
>`maxSeqLen` defines the maximum length of the sequences generated to train the NN.